### ISSUE TITLE
Bug in random_walk

### ISSUE THREAD

**Issue by g-morishita** [rusty1s/pytorch_cluster#111]
Hi, 

If I tried to use `torch_cluster.random_walk`, I encountered the Segmentation fault.

## code to reproduce the bug
Took this code from https://github.com/rusty1s/pytorch_cluster/issues/81 
```python
>>> import torch
>>> from torch_cluster import random_walk
>>> row = torch.tensor([21,21,27,29,54,54,60,62])
>>> col = torch.tensor([27,29,21,21,60,62,54,54])
>>> start = torch.arange(66)
>>> walk = random_walk(row, col, start, walk_length=5)
Segmentation fault (core dumped)
```

## Environment
I use docker environment on macOS Catalina 10.15.7.

Here is the Dockerfile to build a container.

```Dockerfile
FROM continuumio/miniconda3

ENV OPENBLAS_NUM_THREADS 1
ENV DEBIAN_FRONTEND=noninteractive
ENV LANG=ja_JP.UTF-8 

RUN apt-get -y update && apt-get install -y --no-install-recommends \
         gcc \
         wget \
         nginx \
         ca-certificates \
         make \
         git \
         curl \
         xz-utils \
         file \
         sudo \
         patch \
         swig \
    && rm -rf /var/lib/apt/lists/*

RUN conda create -n gnn python==3.8.3 
SHELL ["conda", "run", "-n", "gnn", "/bin/bash", "-c"]
RUN conda install gcc_linux-64 gxx_linux-64
RUN conda clean -a
RUN conda install pytorch torchvision -c pytorch
RUN conda install pytorch-geometric -c rusty1s -c conda-forge
```

## version
```bash
$ conda list | grep torch
ffmpeg                    4.3                  hf484d3e_0    pytorch
pytorch                   1.9.0           py3.8_cuda10.2_cudnn7.6.5_0    pytorch
pytorch-cluster           1.5.9           py38_torch_1.9.0_cu102    rusty1s
pytorch-geometric         1.7.2           py38_torch_1.9.0_cu102    rusty1s
pytorch-scatter           2.0.8           py38_torch_1.9.0_cu102    rusty1s
pytorch-sparse            0.6.12          py38_torch_1.9.0_cu102    rusty1s
pytorch-spline-conv       1.2.1           py38_torch_1.9.0_cu102    rusty1s
torchvision               0.10.0               py38_cu102    pytorch
```

Thank you in advance.

**Comment by rusty1s** [comment]
Fixable by specfiying `num_nodes`:
```
walk = random_walk(row, col, start, walk_length=5, num_nodes=67)
```

**Comment by g-morishita** [comment]
Thank you for the reply. It worked. I will close it.

**Comment by g-morishita** [comment]
For Someone like me, is it a good idea to give a warning message when you do not specify `num_nodes` to run `random_walk`? If you think so, I'll create PR.

**Comment by rusty1s** [comment]
I think a more reasonable solution is to [infer `num_nodes`](https://github.com/rusty1s/pytorch_cluster/blob/master/torch_cluster/rw.py#L34-L35) via:
```python
if num_nodes is None:
        num_nodes = max(int(row.max()), int(col.max()), int(start.max())) + 1
```
in case it is not given.

**Comment by g-morishita** [comment]
The solution sounds good! I created PR.

**Comment by rusty1s** [comment]
Already merged :) Thanks!

**Comment by alessiamarcolini** [comment]
Hey @rusty1s, do you have a timeline for a release with this patch? Thank you 🙏

**Comment by rusty1s** [comment]
Thanks for the reminder. It's indeed time for a release. I will work on it. In the meantime, you can install the latest version via
```
pip install git+https://github.com/rusty1s/pytorch_cluster.git
```

**Comment by rusty1s** [comment]
Finally released [`torch-cluster==1.6.0`](https://github.com/rusty1s/pytorch_cluster/releases/tag/1.6.0).
